src/main/java/org/jenkinsci/plugins/github/pullrequest/GitHubPRMessage.java

```diff
@@ -33,48 +38,61 @@ public GitHubPRMessage(String content) {
     }
 
-    public String expandAll(AbstractBuild<?, ?> build, TaskListener listener) {
-        String content = getContent();
+    /**
+     * Expand all what we can from given run
+     */
+    @Restricted(NoExternalUse.class)
+    @CheckForNull
+    public String expandAll(Run<?, ?> run, TaskListener listener) throws IOException, InterruptedException {
+        return expandAll(getContent(), run, listener);
+    }
+
+    /**
+     * Expand all what we can from given run
+     */
+    @Restricted(NoExternalUse.class)
+    @CheckForNull
+    public static String expandAll(String content, Run<?, ?> run, TaskListener listener)
+            throws IOException, InterruptedException {
         if (content == null || content.length() == 0) {
             return content; // Do nothing for an empty String
         }
-//        // Expand environment variables
-//        String s = build.getEnvironment(listener).expand(content);
-        // Expand build variables
-        content = Util.replaceMacro(content, build.getBuildVariableResolver());
 
-        Jenkins jenkins = Jenkins.getInstance();
-        try {
-            ClassLoader uberClassLoader = Jenkins.getInstance().pluginManager.uberClassLoader;
+        // Expand environment variables
+        String body = run.getEnvironment(listener).expand(content);
 
-            if (jenkins.getPlugin("token-macro") != null) {
-                List macros = null;
+        // Expand build variables + token macro if they available
+        if (run instanceof AbstractBuild<?, ?>) {
+            final AbstractBuild<?, ?> build = (AbstractBuild<?, ?>) run;
+            body = Util.replaceMacro(body, build.getBuildVariableResolver());
 
-                //get private macroses like groovy template ${SCRIPT} if available
-                if (jenkins.getPlugin("email-ext") != null) {
-                    Class<?> contentBuilderClazz = uberClassLoader.loadClass("hudson.plugins.emailext.plugins.ContentBuilder");
-                    Method getPrivateMacrosMethod = contentBuilderClazz.getDeclaredMethod("getPrivateMacros");
-                    macros = new ArrayList((Collection) getPrivateMacrosMethod.invoke(null));
-                }
+            try {
+                Jenkins jenkins = Jenkins.getActiveInstance();
+                ClassLoader uberClassLoader = jenkins.pluginManager.uberClassLoader;
+                List macros = null;
+                if (jenkins.getPlugin("token-macro") != null) {
+                    // get private macroses like groovy template ${SCRIPT} if available
+                    if (jenkins.getPlugin("email-ext") != null) {
+                        Class<?> contentBuilderClazz = uberClassLoader.loadClass("hudson.plugins.emailext.plugins.ContentBuilder");
+                        Method getPrivateMacrosMethod = contentBuilderClazz.getDeclaredMethod("getPrivateMacros");
+                        macros = new ArrayList((Collection) getPrivateMacrosMethod.invoke(null));
+                    }
```

Comment:
I understand the reason of such magic, but it causes much risks when the plugins get updated.
No strong opinion